numpy/polynomial/_polybase.py

```diff
@@ -283,10 +315,82 @@ def __repr__(self):
         name = self.__class__.__name__
         return f"{name}({coef}, domain={domain}, window={window})"
 
+    def __format__(self, fmt_str):
+        if fmt_str == '':
+            return self.__str__()
+        if fmt_str not in ('ascii', 'unicode'):
+            raise TypeError(
```

Comment:
Also should be `ValueError`.  Python uses `ValueError`, e.g.
```
In [14]: f"{100:q}"
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-14-f0e76aaf133d> in <module>
----> 1 f"{100:q}"

ValueError: Unknown format code 'q' for object of type 'int'

In [15]: f"{100:ABC}"
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-15-188fbe0b055e> in <module>
----> 1 f"{100:ABC}"

ValueError: Invalid format specifier
```

Reply:
Nice catch, thanks for pointing this out!